GNUmakefile

```diff
@@ -52,18 +52,20 @@ test-compile:
 	fi
 	go test -c $(TEST) $(TESTARGS)
 
-dependencies:
-	@echo "Download go.mod dependencies"
-	@go mod download
-
-install-tools: dependencies
+BIN=$(CURDIR)/bin
+$(BIN)/%:
 	@echo "Installing tools from tools/tools.go"
-	@cat tools/tools.go | grep _ | awk -F '"' '{print $$2}' | xargs -tI {} go install {}
+	@cat tools/tools.go | grep _ | awk -F '"' '{print $$2}' | GOBIN=$(BIN) xargs -tI {} go install {}
+
+# Inject ./bin into PATH to allow scripts/generate-docs.go to access local tfplugindocs binary
+generate-docs: $(BIN)/tfplugindocs
+	PATH=$(PATH):$(BIN) go run scripts/generate-docs.go
 
-generate-docs: install-tools
-	go run scripts/generate-docs.go
+validate-docs: $(BIN)/tfplugindocs
+	$(BIN)/tfplugindocs validate
```

Comment:
I don't believe so, as it's calling the binary from its relative filepath instead of letting the shell look it up on the system `PATH`, but I might be wrong. In the case of `generate-docs` the script calls it without specifiying the location so the `PATH` is needed there. 

Now you mention it, it could be cleaner to specify `./bin/tfplugindocs` inside the script to avoid modifying the `PATH` for that command, if we decide that a non-global install is the right way to go. Or if could be nice to leave it configurable from the Makefile too so the location is saved in one place, in which case maybe passing an argument in could be best.

Reply:
Just a heads up here - I've pushed in a change to avoid modifying `PATH` for the `generate-docs`